### PR TITLE
Initial setup

### PR DESCRIPTION
**Pull request by SaraOsmanovic**
The initial setup of the repository has been copied over with slight changes with respect to requests made.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,207 @@
...
+		- [Quickstart](#quickstart)
+		- [Repositories](#repositories)
+		- [Branches](#branches-1)
+		- [Issues](#issues-1)
+		- [Bonus Documentation](#bonus-documentation)
+
+# Documentation and resources on how to use GitHub for Project 3
+## Getting Started with the Project
+
+This is a group project. As such, you're expected to equally contribute to all parts of the project. To make the collaboration easier for you, we have moved this project entirely to GitHub. However, to fully leverage the features that GitHub offers, you have to be familiar with what they are and how to use them.
```
Comment (kapooramanpreet) [inline README.md]: Change this: "To make the collaboration easier for you, we have moved this project entirely to GitHub. However, to fully leverage the features that GitHub offers, you have to be familiar with what they are and how to use them" to "To ensure everyone is contributing, we have moved this project entirely to GitHub and will use GitHub logs to understand your contributions. To fully leverage the features that GitHub offers, you have to be familiar with what they are and how to use them".

File: README.md [hunk 2]
```diff
@@ -1 +1,207 @@
...
+
+The goal of this project is to get you familiar with the creative process of software development for a general purpose use and how to use data structures or algorithms for those applications. To go with the professional development that the project offers in terms of general use application, you will also get more familiar with version control and GitHub platform to facilitate the collaborative work with extensive planning and organization abilities.
+
+This step by step process if linked in this video:
```
Comment (kapooramanpreet) [inline README.md]: typo - if -> is

Also, add: This step by step process is linked in this video **if you aren't familiar with GitHub**:

File: README.md [hunk 3]
```diff
@@ -1 +1,207 @@
...
+- [YouTube](https://youtu.be/CaGUAi6jnik) - Doesn't have chapters but has subtitles
+- [Loom](https://www.loom.com/share/5f06c23105cf449a8f526b997cb3768b?sid=e30f822f-11c1-4ce7-8821-008932fd930a) - Has chapters and transcript
+
```
Comment (kapooramanpreet) [inline README.md:44]: Add: Here are the instructions on how to Collaborate for Project 3 **which are similar to what was covered in the video.** You may skip reading these if you have watched the video:

File: README.md [hunk 4]
```diff
@@ -1 +1,207 @@
...
+### Step 1: Create Your Project Repository
+
+This is a template repository. If you want to know more about those, feel free to follow [this link](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-template-repository) to the GitHub documenation about templates and how to create them. You've used templates before for projects 1 and 2, and you will use it now, too. 
+
+The first step of your project is to create the project repository that will host your source code and documentation.
```
Comment (kapooramanpreet) [inline README.md]: Add: "**Only one member of the team needs to create a repo and everyone else copies that repo using git clone**".

File: README.md [hunk 5]
```diff
@@ -1 +1,207 @@
...
+1. Navigate to the main page of the repository.
+2. Above the file list, in line with the repository name, click on the big green buton saying **"Use this template"**
```
Comment (kapooramanpreet) [inline README.md]: - [x] typo button
- [x] template repo is not showing up

File: README.md [hunk 6]
```diff
@@ -1 +1,207 @@
...
+3. Select **"Create a new repository"** (This step is important. Do not open in codespace. Make your own repository for your own project.)
```
Comment (kapooramanpreet) [inline README.md:52]: reframe: Make your own repository for your own project.

File: README.md [hunk 7]
```diff
@@ -1 +1,207 @@
...
+
+#### Commits
+
+Commits are a way to update the repository with the changes you made. Each commit has information about who made the change, what changes were made, when the change was made, and a message that describes the changes (and/or if it refers to an [Issue](#issues)).
+
+Commits are very important part of collaboration and should follow an agreed on convention for the team. Each commit also offers a way to organize the changes because it shows a very short concise description of the change made. As such, make sure to commit frequently and that each commit solves only one issue. As with Issues, your commits should be atomic (contains only one change).
+
+##### Guidelines for good commit messages
```
Comment (kapooramanpreet) [inline README.md:104]: give an example in each category if possible.

File: README.md [hunk 8]
```diff
@@ -1 +1,207 @@
...
+
+Once you've made commits and are done working on the feature your branch was for, you should make a pull request. A pull request is meant to allow your teammates to review your changes, give you feedback, and the merge all the functional code into one central location (main branch).
+
+## README, Markdown, and Report Submission
+
+For this project, you're expected to submit a report that is similar to those you've done for projects 1 and 2, with the proposal submitted in 3a included. To centralize this with your source code and video, you will write your Report in Markdown as a README file. Since you're supposed to create your project repository from this template, edit the existing README.md file, remove all the contents that are currently there, and replace them with your report.
+
+Putting the README.md in your root or docs directory will automatically be recognized by GitHub and will be shown to anyone who visits the repository. For all documents, including README files, commit messages and issues, GitHub uses Markdown for formatting.
+
+### Important Elements
```
Comment (kapooramanpreet) [inline README.md]: All of the markdown stuff should be its own readme file.
I think we can have that "resources folder" and within that you have markdown file as well as images folder

File: README.md [hunk 9]
```diff
@@ -1 +1,207 @@
...
+
+Once you've made commits and are done working on the feature your branch was for, you should make a pull request. A pull request is meant to allow your teammates to review your changes, give you feedback, and the merge all the functional code into one central location (main branch).
+
+## README, Markdown, and Report Submission
+
+For this project, you're expected to submit a report that is similar to those you've done for projects 1 and 2, with the proposal submitted in 3a included. To centralize this with your source code and video, you will write your Report in Markdown as a README file. Since you're supposed to create your project repository from this template, edit the existing README.md file, remove all the contents that are currently there, and replace them with your report.
```
Comment (kapooramanpreet) [inline README.md]: Should we provide a template for the report if we are enforcing markdown? Thoughts

File: README.md [hunk 10]
```diff
@@ -1 +1,207 @@
...
+	\[link](www.example.com)
+
+with the bulleted lists:
+
+	- \[Section 1](#section-1)
+
+
+If you have VS Code, it will then automatically add all headings into the table upon saving. If you want to exclude the title and table of contents as headings use the HTML heading syntax (\<h1>Heading 1\</h1>, \<h2>Heading 2\</h2>, \<h3>Heading 3\</h3>, ...)
+
+## Additional Resources and Documentation:
```
Comment (kapooramanpreet) [inline README.md:144]: this can stay here.

**Review by kapooramanpreet — CHANGES_REQUESTED** [review]
This is a good draft Sara. I think this is missing two things and we should make Github mandatory only after we fix the two:
1. adding an example on local development and how it connects with remote on GitHub
2. an explicit short example of the process where you and two other members go through the workflow.
Lets discuss this more and thanks for creating this :)

File: README.md (hunk 7 above)
Comment (SaraOsmanovic) [reply README.md:104]: I added an example of commit `Feat: implement user input parsing for calculator to split string into tokens (Issue #1)`. Would this be enough or did you mean for me to add examples for other parts of the README, too?

File: README.md (hunk 9 above)
Comment (SaraOsmanovic) [reply README.md]: I will make a template report for them to use in their project.

File: README.md (hunk 8 above)
Comment (SaraOsmanovic) [reply README.md]: Made separate directory for Markdown info (linked it in the README) and put the template repo in there, too.

File: README.md (hunk 10 above)
Comment (SaraOsmanovic) [reply README.md:144]: I put it in the resources/MARKDOWN.md file. Is that okay or should I copy it over to the README?

**Pull request merged**